MSAL/src/MSALAccount.m

```diff
@@ -0,0 +1,194 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALAccount.h"
+#import "MSIDClientInfo.h"
+#import "MSIDAccount.h"
+#import "MSALAccount+Internal.h"
+#import "NSURL+MSIDExtensions.h"
+#import "MSIDAuthority.h"
+#import "MSIDAADV2IdTokenClaims.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDAccount.h"
+#import "MSALAccountId+Internal.h"
+
+@interface MSALAccount ()
+
+@property (nonatomic) MSALAccountId *homeAccountId;
+@property (nonatomic) NSString *displayableId;
+@property (nonatomic) NSString *environment;
+
+@end
+
+@implementation MSALAccount
+
+- (id)initWithDisplayableId:(NSString *)displayableId
+                       name:(NSString *)name
+              homeAccountId:(NSString *)homeAccountId
+             localAccountId:(NSString *)localAccountId
+                environment:(NSString *)environment
+                   tenantId:(NSString *)tenantId
+                 clientInfo:(MSIDClientInfo *)clientInfo
+{
+    self = [super init];
+
+    if (self)
+    {
+        _displayableId = [displayableId copy];
+        _name = [name copy];
+        _homeAccountId = [homeAccountId copy];
+        _localAccountId = [localAccountId copy];
+        _environment = [environment copy];
+
+        NSString *uid = clientInfo.uid;
+        NSString *utid = clientInfo.utid;
```

Comment:
Just curious, why do we need to copy a string given that its content is not changeable? :)

Reply:
You're correct, those are immutable strings and we don't need to copy. Fixed.